R/assertions.R

```diff
@@ -368,25 +369,33 @@ assert_logical_scalar <- function(arg, optional = FALSE,
 #' try(example_fun(dm, "USUBJID"))
 #'
 #' try(example_fun(dm, toupper(PARAMCD)))
-assert_symbol <- function(arg, optional = FALSE) {
+assert_symbol <- function(arg,
+                          optional = FALSE,
+                          arg_name = rlang::caller_arg(arg),
+                          message = NULL,
+                          class = "assert_symbol",
+                          call = parent.frame()) {
   assert_logical_scalar(optional)
 
   if (optional && is.null(arg)) {
     return(invisible(arg))
   }
 
-  if (is_missing(arg)) {
-    err_msg <- sprintf("Argument `%s` missing, with no default", arg_name(substitute(arg)))
-    abort(err_msg)
-  }
+  # set default error message, if not specified
+  message <-
+    message %||%
+    ifelse(
+      is_missing(arg),
+      "Argument {.arg {arg_name}} must be a {.cls symbol}, but is missing.",
+      "Argument {.arg {arg_name}} must be a {.cls symbol}, but is {.obj_type_friendly {arg}}."
+    )
 
-  if (!is.symbol(arg)) {
-    err_msg <- sprintf(
-      "`%s` must be a symbol but is %s",
-      arg_name(substitute(arg)),
-      what_is_it(arg)
+  if (is_missing(arg) || !is.symbol(arg)) {
+    cli::cli_abort(
```

Comment:
can we use https://github.com/pharmaverse/admiraldev/blob/main/R/admiraldev-package.R instead of the `::` just to be consistent with previous work?